api/specs.proto

```diff
@@ -409,8 +409,40 @@ message SecretSpec {
 	// - golang: Go templating
 	Driver templating = 3;
 
-	// Driver is the the secret driver that is used to store the specified secret
+	// Driver is the the secret driver that is used to store the specified secret.  One example
+	// is a CFSSL driver for CA type secrets; the type will be "cfssl", and the options will
+	// contain the URL for the CFSSL remote REST API-based signer.
 	Driver driver = 4;
+
+	enum SecretType {
+		// BASIC represents the basic secret data type - this is the default secret type
+		BASIC = 0;
+
+		// CA represents a CA certificate and private key pair
+		CA = 1;
+
+		// LEAFCERT represents a leaf certificate and privat ekey pair
+		LEAFCERT = 2;
+	}
```

Comment:
If we have the `oneof` (https://github.com/docker/swarmkit/pull/2331/files#r129461497), I think this might not be necessary. There could be a type switch on the `oneof` value instead.